### PR TITLE
Proximity fixes

### PR DESCRIPTION
**Pull request by HypnoMama**
- refactor to remove multiple getProximityData calls on mapClick vs address search - now only one call is made in MapContainer, removed call from AddMarker
- proximityData call is now cancelled on multiple address search/map clicks
- remove loading from SiteInformation page and move to AddOpportunity -> on continue click if proximity call is not finished page will show a Loading Screen modal before routing to next page

### CODE REVIEW

File: cit3.0-web/src/components/AddMarker/ResourceMarker.js [hunk 1]
```diff
@@ -27,5 +27,6 @@ export default function ResourceMarker({ resourceName, resources }) {
 }
 ResourceMarker.propTypes = {
   resourceName: PropTypes.string.isRequired,
-  resources: PropTypes.arrayOf(PropTypes.shape).isRequired,
+  resources: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()])
```
Comment (JustinMacaulay) [inline cit3.0-web/src/components/AddMarker/ResourceMarker.js:30]: does this still count as an ArrayOf or was that part not needed?

File: cit3.0-web/src/components/MapContainer/MapContainer.js [hunk 2]
```diff
@@ -1,48 +1,63 @@
...
-  const run = async () => {
-    const soilData = await getSoilAndElevationData(coords);
+  const { CancelToken } = axios;
+  let source = CancelToken.source();
+
+  const run = async (sourceToken) => {
+    // proximity call is running, do not unmount component if user hits continue
+    setProximityInProgress(true);
+    console.log("running");
+    const [soilData, proximityData] = await axios.all([
```
Comment (JustinMacaulay) [inline cit3.0-web/src/components/MapContainer/MapContainer.js:33]: I would prefer that the axios call be in an actions file or something like that. I just like to keep all of the API calls together.

**Review by JustinMacaulay — COMMENTED** [review]
Just a few comments but otherwise looks good.

File: cit3.0-web/src/components/AddMarker/ResourceMarker.js (hunk 1 above)
Comment (HypnoMama) [reply cit3.0-web/src/components/AddMarker/ResourceMarker.js:30]: That part was old and outdated

**Review by zacpez — APPROVED** [review]

**Pull request merged**